robot_smach_states/src/robot_smach_states/util/designators/checks.py

```diff
@@ -35,14 +36,24 @@ def check_resolve_type(designator, *allowed_types):
     """
 
     if isinstance(designator.resolve_type, list):
-        real_resolve_type = designator.resolve_type[0]
-        real_allowed_type = allowed_types[0][0] #allowed_types is a list (because of the *).
+        real_resolve_type = designator.resolve_type[0]  # Designator should only resolve to one type.
+        correct_resolve_type = False
+        for allowed_type in allowed_types:  # allowed_types is a tuple (because of the *).
+            real_allowed_type = allowed_type[0] if isinstance(allowed_type, list) else allowed_type
+
+            if real_resolve_type == real_allowed_type:
+                correct_resolve_type = True
+                break
 
-        if not real_resolve_type == real_allowed_type:
-            raise TypeError("{0} resolves to {1} but should resolve to one of {2}".format(designator, designator.resolve_type, allowed_types))
+        if not correct_resolve_type:
+            raise TypeError("{0} resolves to {1} but should resolve to one of {2}".format(designator,
```

Comment:
Why not raise directly from line 39, instead of breaking?

Reply:
Did you check after my last commit(511baf4a6b78e311b7aab5ea645343fe1130e3d6)?

Reply:
I did, but why split and set that bool and then read it later?

Reply:
Because you need to check all allowed types for a good match.